Lib/socketserver.py

```diff
@@ -273,18 +270,16 @@ def service_actions(self):
     # - finish_request() instantiates the request handler class; this
     #   constructor will handle the request all by itself
 
+    def _get_timeout(self):
+        """Hook so child classes can support other sources of timeout."""
+        return self.timeout
+
     def handle_request(self):
         """Handle one request, possibly blocking.
 
         Respects self.timeout.
         """
-        # Support people who used socket.settimeout() to escape
-        # handle_request before self.timeout was available.
-        timeout = self.socket.gettimeout()
-        if timeout is None:
-            timeout = self.timeout
-        elif self.timeout is not None:
-            timeout = min(timeout, self.timeout)
```

Comment:
Wouldn't this affect users that directly inherit from `BaseServer` instead?

Reply:
The new version I just pushed leaves the existing timeout logic in place, but uses `hasattr` to make it safe for any subclasses that don't have a `socket` attribute. I can put it back to the way it was if you think that's still too much, but this should be less obtrusive while still increasing the range of possibilities for users of the base class.

Reply:
Maybe the NEWS entry should be moved out of the documentation category at this point?